modules/guides/guides.routing.yml

```diff
@@ -7,8 +7,8 @@ guides.list:
     _permission: 'access guides'
 
 guides.guide:
-  path: '/admin/guides/{filename}'
+  path: '/admin/guides/{subdirectory}/{filename}'
```

Comment:
What is the benefit of separating the directory from the filename? Would it be possible to use `/admin/guides/{path}` where `{path} would be a relative path to a file within the guides directory (without file extension)?

Reply:
I think, yes, it would be possible, but this solution didn't come to my mind. Separating the file path into two parameters seems more logical to me, because in this way the path is more dynamic, and it's easier to reuse these parameters in the Controller than the abbreviated path without the file extension. I know that now I use these two parameters together everywhere, but maybe later someone will need only the file name or only the subdirectory.

Reply:
> maybe later someone will need only the file name or only the subdirectory.
that could be easily figured our from the `{path}` as well, `basename($path)` for example.

Reply:
Ok, then I don't know the answer. But there should be something, because you started the question with "What is the benefit" and not with "Is there any benefit". :D

Reply:
grammar natzy++ :P I wanted to force you to explain why did you choose this solution because I may not see why it is better than a simple `{path}` that contains everything except the file extension. Because you split the path to two pieces you always have to build the full path whenever you need it. If your only explanation that someone may need only the filename that is not good enough for me because filename can be retrieved from the `{path}` easily.

Reply:
Ok, after trying out {path} and replacing $subdirectory and $file_name with $path everywhere, I found a much better reason why it is better to separate the path into two parameters: it doesn't work with {path} and it always throws an InvalidParameterException, because "Route parameters can contain any values except the / slash character, because that's the character used to separate the different parts of the URLs."

Reply:
But if you don't accept my reason, that someone may need only the filename, then why should I accept your reason that I always have to build the full path whenever I need it? If I don't split the path I still have to build the full path, and to build the full path from four parts is not more complicated than build it from three parts. :P